owners/test/local_repo.test.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const {LocalRepository} = require('../src/local_repo');
+const sinon = require('sinon');
+const fs = require('fs').promises;
+
+
+describe('local repository', () => {
+  const sandbox = sinon.createSandbox();
+  let repo;
+
+  beforeEach(() => {
+    repo = new LocalRepository('path/to/repo');
+    sandbox.stub(repo, 'getAbsolutePath').callsFake((relativePath) => {
+      return `path/to/repo/${relativePath}`;
+    });
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  describe('constructor', () => {
+    it('initializes the root directory and remote', () => {
+      repo = new LocalRepository('path/to/repo', 'my_remote');
+      expect(repo.rootDir).toEqual('path/to/repo');
+      expect(repo.remote).toEqual('my_remote');
+    });
+
+    it('defaults to "origin" remote', () => {
+      expect(repo.remote).toEqual('origin');
+    });
+  });
+
+  describe('checkout', () => {
+    beforeEach(() => {
+      sandbox.stub(repo, 'runCommands');
```

Comment:
Whenever possible, don't stub anything in the class that you're testing. Stub the dependencies instead

Reply:
Leaving a `TODO` here for now. Since the `local_repo` module uses `util.promisify(child_process.exec)`, which is assigned and keeps a reference to the original `exec` in a scope, stubbing is a bit tricky and will either be hacky or require some reworking of the `requires` in `local_repo`